reasoning_gym/algorithmic/pool_matrix.py

```diff
@@ -0,0 +1,114 @@
+"""Perform average / max pooling on a matrix"""
+
+from copy import deepcopy
+from dataclasses import dataclass
+from random import Random
+from typing import Dict, Optional
+
+import numpy as np
+
+from ..factory import ProceduralDataset, register_dataset
+
+QUESTION_TEMPLATE = """Perform {pool_type} pooling on the following matrix:
+{matrix}
+"""
+
+
+@dataclass
+class PoolMatrixConfig:
+    """Configuration for Pool Matrix dataset generation"""
+
+    max_rows: int = 10  # Maximum rows of the matrix
```

Comment:
Could we add `min_rows` & `min_cols` both with default values 2? If it is smaller than 2x2 it doesn't really feel like a matrix ;-).